lib/ansible/modules/windows/win_iis_webbinding.ps1

```diff
@@ -1,131 +1,454 @@
 #!powershell
 
-# (c) 2015, Henrik Wallström <henrik@wallstroms.nu>
-#
-# This file is part of Ansible
-#
-# Ansible is free software: you can redistribute it and/or modify
-# it under the terms of the GNU General Public License as published by
-# the Free Software Foundation, either version 3 of the License, or
-# (at your option) any later version.
-#
-# Ansible is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License
-# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
-
-
-# WANT_JSON
-# POWERSHELL_COMMON
-
-$params = Parse-Args $args;
-
-$name = Get-AnsibleParam $params -name "name" -failifempty $true
-$state = Get-AnsibleParam $params "state" -default "present" -validateSet "present","absent"
-$host_header = Get-AnsibleParam $params -name "host_header"
-$protocol = Get-AnsibleParam $params -name "protocol"
-$port = Get-AnsibleParam $params -name "port"
-$ip = Get-AnsibleParam $params -name "ip"
-$certificatehash = Get-AnsibleParam $params -name "certificate_hash" -default $false
-$certificateStoreName = Get-AnsibleParam $params -name "certificate_store_name" -default "MY"
+# Copyright: (c) 2017, Noah Sparks <nsparks@outlook.com>
+# Copyright: (c) 2015, Henrik Wallström <henrik@wallstroms.nu>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
-$binding_parameters = @{
-  Name = $name
-};
+#Requires -Module Ansible.ModuleUtils.Legacy
+
+$params = Parse-Args -arguments $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+
+$name = Get-AnsibleParam $params -name "name" -type str -failifempty $true -aliases 'website'
+$state = Get-AnsibleParam $params "state" -default "present" -validateSet "present","absent"
+$host_header = Get-AnsibleParam $params -name "host_header" -type str
+$protocol = Get-AnsibleParam $params -name "protocol" -type str -default 'http'
+$port = Get-AnsibleParam $params -name "port" -default '80'
+$ip = Get-AnsibleParam $params -name "ip" -default '*'
+$certificateHash = Get-AnsibleParam $params -name "certificate_hash" -type str -default ([string]::Empty)
```

Comment:
We usually just omit the `-default` part if there is no default. That way you only need to check if `$certificateHash` is `$null` or not to determine whether it was set.

Reply:
Thanks for checking it out. It looks like I did this because the binding object returned by get-webbinding puts empty strings in certhash/certstore rather than null so comparing to $null wasn't working. I can probably just do if ($string) instead and get rid of that default to make it cleaner.

I'll test it soon.

Reply:
Cool, if you need it as empty that's fine I just wasn't sure if you set that on purpose.

Reply:
Ok. If it's alright to leave it, I'd like to.

Reply:
Sound good to me.